src/streamlink/plugins/dogus.py

```diff
@@ -43,8 +49,17 @@ def _get_streams(self):
 
         # http://eurostartv.com.tr/canli-izle
         dd_script = root.xpath("string(.//script[contains(text(), '/live/hls/')][1]/text())")
+        dd_script2 = root.xpath("string(.//script[contains(text(), 'daionUrl')][1]/text())")
         if dd_script:
             m = self._re_live_hls.search(dd_script)
+            if m and "startv" in m.group(1):
+                return HLSStream.parse_variant_playlist(self.session, m.group(1).replace("//", "http://"))
+            elif m:
+                return HLSStream.parse_variant_playlist(self.session, m.group(1))
+        # https://www.tlctv.com.tr/canli-izle
+        # https://www.dmax.com.tr/canli-izle
+        elif dd_script2:
+            m = self._re_live_hls2.search(dd_script2)
             if m:
                 return HLSStream.parse_variant_playlist(self.session, m.group(1))
```

Comment:
`update_scheme` should be used for setting missing explicit schemes. `force=False` will only set it if it's missing, and since the site doesn't support TLS, `"http://"` needs to be the first argument.

Also don't change and mix up the existing regex.

----

```py
from streamlink.utils.url import update_scheme
```

```py
        # http://eurostartv.com.tr/canli-izle
        dd_script = root.xpath("string(.//script[contains(text(), '/live/hls/')][1]/text())")
        if dd_script:
            m = self._re_live_hls.search(dd_script)
            if m:
                return HLSStream.parse_variant_playlist(self.session, update_scheme("http://", m.group(1), force=False))

        # https://www.tlctv.com.tr/canli-izle
        # https://www.dmax.com.tr/canli-izle
        dd_script2 = root.xpath("string(.//script[contains(text(), 'daionUrl')][1]/text())")
        if dd_script2:
            m = self._re_live_hls2.search(dd_script2)
            if m:
                return HLSStream.parse_variant_playlist(self.session, m.group(1))
```

Reply:
`_re_live_hls = re.compile(r"[\"|']((?:https?:)?//[^']+/live/hls/[^']+?)[\"|']")`

starttv.com.tr doesn't work without [\"|']. However you can find a better regex. I am new streamlink user.